### PR TITLE
Master - TimeAccounting port to OM

### PR DESCRIPTION
**Pull request by s7design**
Hi @carlosfrodriguez 

Here is ported TimeAccounting to OM. 

We also change the following tings:

* in TimeAccounting.xml year is changed to 2015 TimeAccounting::DefaultUserDateStart and TimeAccounting::DefaultUserDateStart. Is it ok?
* update to modern input fields. We have problem with jquery-combobox.js. After adding Morenize class and deleting Autocompletion: [% Data.EnableAutocompletion | html %] from AgentTimeAccountingEdit.tt, these specific select box fields work fine. I think that TimeAccounting.Agent.EditTimeRecords should be cleaned up because some Init function for dropdown is not needed anymore. 
We had to change _ProjectList function in AgentTimeAccountingEdit.pm. Modern select box works different then the previous one. All projects are displayed two times.
screen shot 2015-08-04 at 3 40 43 pm

* we fixed filters in AgentTimeAccountingSetting.tt like it is solved in the Framework

There is one more issue with width of input fields in Widget action. Dropdown fields in sidebar is wider than the other ones.
screen shot 2015-08-04 at 4 08 40 pm

The same in Admin Customer screens:
screen shot 2015-08-04 at 4 06 43 pm

I will be able to solve this for the Framework, and automatically it will be solved here. 

I know that there are many changes in the one PR, but only with all of these changes, TimeAccounting master is able to work properly now.
If you have any suggestion how we can help you maybe with splitting in more PRs or with additional improvement please let me know.

Regards
Zoran

### CODE REVIEW

File: Kernel/Modules/AgentTimeAccountingEdit.pm [hunk 1]
```diff
@@ -1329,30 +1341,6 @@ sub _ProjectList {
         %{ $Self->{LastProjectsRef} } = map { $_ => 1 } @LastProjects;
     }
 
-    PROJECTID:
-    for my $ProjectID (
-        sort { $Project{Project}{$a} cmp $Project{Project}{$b} }
-        keys %{ $Project{Project} }
-        )
-    {
-        next PROJECTID if !$Self->{LastProjectsRef}->{$ProjectID};
```
Comment (carlosfrodriguez) [inline Kernel/Modules/AgentTimeAccountingEdit.pm]: Hi @s7design,

This is in fact important, as it creates the list of last user projects to be at the top of the list, for sure there are some duplicates, but it is very fast to get the projects you use most.

((enjoy))
cr

File: Kernel/Modules/AgentTimeAccountingEdit.pm [hunk 2]
```diff
@@ -1329,30 +1341,6 @@ sub _ProjectList {
...
-            Value => $Project{Project}{$ProjectID},
-        );
-        push @List, \%Hash;
-
-        # at the moment it is not possible mark the selected project
-        # in the favorite list (I think a bug in Build selection?!)
-    }
-
-    # add the separator
-    push @List, {
```
Comment (carlosfrodriguez) [inline Kernel/Modules/AgentTimeAccountingEdit.pm]: Hi @s7design,

It si possible to add disabled items in modern input fields?


((enjoy))
cr

**Comment by carlosfrodriguez** [comment]
Hi @s7design, Zoran,

I think we should get another approach with this, I think it will be better to have a clean porting, and after that another pull request to try to use modern inputs then we can track and try to fix all the issues, and if is not possible just rollback that part.

((enjoy))
cr

**Comment by s7design** [comment]
Hi @carlosfrodriguez 

I updated this PR. I roll-backed to commit where is solved only porting to OM.
After you merge this PR I will be able to update to modern fields. 

Regards
Zoran

**Comment by carlosfrodriguez** [comment]
Hi @s7design, Zoran,

This pull request is merged in https://github.com/OTRS/TimeAccounting/commit/28dd8caeea1b9165be8feeb0368dbd4f4b0d1c89, With some code style cleanup. AutoComplete was temporary disabled in AgentTimeAccoutningEdit.tt to make the package usable.

((enjoy))
cr

**Pull request closed without merging**